server/task/task-memory/src/test/java/org/apache/james/task/SerialTaskManagerWorkerTest.java

```diff
@@ -169,6 +170,38 @@ void theWorkerShouldReportThatATaskIsInProgress() throws InterruptedException {
         latch.countDown();
     }
 
+    @Disabled("JAMES-3172 We cannot cancel computation started by Reactor")
+    @Test
+    void reactiveTasksShouldBeCancellable() throws InterruptedException {
+        // Provide a task ticking every 100ms in a separate reactor thread
+        AtomicInteger tickCount = new AtomicInteger();
+        int tikIntervalInMs = 100;
+        MemoryReferenceTask tickTask = new MemoryReferenceTask(() -> Flux.interval(Duration.ofMillis(tikIntervalInMs))
+            .flatMap(any -> Mono.fromCallable(() -> {
+                tickCount.incrementAndGet();
+                return Task.Result.COMPLETED;
+            }))
+            .reduce(Task::combine)
+            .thenReturn(Task.Result.COMPLETED)
+            .block());
+
+        // Execute the task
+        TaskId id = TaskId.generateTaskId();
+        TaskWithId taskWithId = new TaskWithId(id, tickTask);
+        Mono<Task.Result> resultMono = worker.executeTask(taskWithId).cache();
+        resultMono.subscribe();
+        Awaitility.waitAtMost(org.awaitility.Duration.TEN_SECONDS)
+            .untilAsserted(() -> verify(listener, atLeastOnce()).started(id));
+
+        worker.cancelTask(id);
+
+        int tikCountSnapshot1 = tickCount.get();
+        Thread.sleep(2 * tikIntervalInMs);
+        int tikCountSnapshot2 = tickCount.get();
+        // If the task had effectively been canceled tikCount should no longer be incremented
+        assertThat(tikCountSnapshot1).isEqualTo(tikCountSnapshot2);
```

Comment:
I don't think we provide this guarantee: how do you know you are able to interrupt a process that fast?

Reply:
Yes I understand it's working at this time on your workstation but I'm more concern about whether we think it's something that should work in general (even with your pet process).

I honestly don't know.

Reply:
CI test runs will tell us.

Worst case we add a delay.

Reply:
CI will tell us for the current reactor version, jdk, etc.
However it's still not a proof it has the work.
Let's say with a comment that this requirement may be too much and is not strictly needed for James to work properly, so that if we start having problem with the test, we don't fight too much before deleting it. ok?